File: ProjetoJava/src/heranca.java
class Animal {
    public void som() {}
}

class Cachorro extends Animal2 {
    public void som() {
        System.out.println("O cachorro faz: Au au!");
    }
}

class Gato extends Animal2 {
    public void som() {
        System.out.println("O gato faz: Miau!");
    }

    public static void main(String[] args) {
        Cachorro2 cachorro = new Cachorro2();
        Gato2 gato = new Gato2();
        cachorro.som();
        gato.som();
    }
}

File: GoLangProject/heranca.go
package main

import "fmt"

type Animal interface {
	Som()
}

type Cachorro struct{}

func (c Cachorro) Som() {
	fmt.Println("O cachorro faz: Au au!")
}

type Gato struct{}

func (g Gato) Som() {
	fmt.Println("O gato faz: Miau!")
}

func main() {
	cachorro := Cachorro{}
	gato := Gato{}

	cachorro.Som()
	gato.Som()
}
